doc/cmd/preprocess.md

```diff
@@ -0,0 +1,15 @@
+# Preprocess command
+
+This command allows you to preprocess your data before passing it to any command you need.
+It converts the input files into Parquet files, after selecting commits, filtering by language, extracting UASTs and possibly taking out some of the fields.
+You can specify the following arguments:
+
+- `-r`/`--repositories` : Path to the input files
+- `--parquet`: If your input files are Parquet files
+- `--graph`: Path to the output Graphviz file if you wish to keep the tree
+- `-o`/`--output`: Path to the output Parquet files
+- `-x`/`--mode`: What to extract from repositories: files, functions or repository itself
+- `-f`/`--fields`: Fields to keep, defaults to all, i.e. "blob_id", "repository_id", "content", "path", "commit_hash" and "uast"
+- `-l`/`--languages` : Languages to keep, defaults to all languages detected by Babelfish
+- `--dzhigurda`: Index of the last commit to keep, defaults to 0 (only the head), 1 is HEAD~1, etc.
```

Comment:
same as in the `args`, it's hard to understand as is: ->
"""
Number of additional snapshots to keep for each document.
By default (`dzhigurda=0`) we only keep the HEAD, with `dzhigurda=1` we keep HEAD and HEAD~1, with `dzhigurda=2` we keep HEAD and HEAD~2, etc. With `dzhigurda=-1` we keep all possible snapshots for each document.
"""

Reply:
I do not like `snapshots` prefer to keep `commits`, but I get your point.